### ISSUE TITLE
phpcs false positive PSR2 error when parenthesis around new object construction follow an if

### ISSUE THREAD

**Issue by still-dreaming-1** [squizlabs/PHP_CodeSniffer#1705]
The following code correctly shows no sniff errors:

```
<?php

namespace Acs\Csst;

class Reproduce
{
    public function someMethod()
    {
        (new \StdClass());
    }
}
```
The following code does show 3 sniff errors:

```
<?php

namespace Acs\Csst;

class Reproduce
{
    public function someMethod()
    {
        if (false) {
        }
        (new \StdClass());
    }
}
```
The errors:

```
FOUND 3 ERRORS AFFECTING 2 LINES                                                                                               
----------------------------------------------------------------------                                                         
 10 | ERROR | [x] Space before opening parenthesis of function call
    |       |     prohibited
 10 | ERROR | [x] Opening parenthesis of a multi-line function call
    |       |     must be the last content on the line
 11 | ERROR | [x] Closing parenthesis of a multi-line function call
    |       |     must be on a line by itself
```

As far as why you would want to enclose a new object in parenthesis, it allows you to call a method of the object on the same line without having to assign it to a variable, but I left that out to show that the reproduce is very simple.

It might help to see, this is what the code looks like after using phpcbf on it:

```
<?php

namespace Acs\Csst;

class Reproduce
{
    public function someMethod()
    {
        if (false) {
        }(new \stdClass());
    }
}
```

After applying those changes, phpcs no longer finds any errors in the file.

**Comment by jrfnl** [comment]
Duplicate of #1669 and #1678 and #1695. This has been fixed in `master` already.

**Comment by gsherwood** [comment]
Closing as duplicate. Bug fix release should be out early next week.
